File: ai-mock/ai/character.py
from ai.point import Point

class Character:
    def __init__(self, id, y, x):
        self.point = Point(y, x)

    def __eq__(self, other):
        return self.point == other.point

    def __hash__(self):
        return self.point.__hash__()

    def __repr__(self):
        return self.point.__repr__()

File: ai/src/character.cpp
#ifndef AI_CHARACTER
#define AI_CHARACTER

#include "character.h"

Character::Character() { };

Character::Character(int _id, int y, int x) {
    point = Point(y, x);
    id = _id;
}

bool Character::operator==(const Character &other) const {
    return this->point == other.point;
}

bool Character::operator<(const Character &other) const {
    if (dist == other.dist ) return id < other.id;
    return dist < other.dist;
}


#endif
